plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/conversions/StreamConversion.kt

```diff
@@ -0,0 +1,38 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.kotlin.nj2k.conversions
+
+import org.jetbrains.kotlin.analysis.api.KtAnalysisSession
+import org.jetbrains.kotlin.nj2k.NewJ2kConverterContext
+import org.jetbrains.kotlin.nj2k.RecursiveConversion
+import org.jetbrains.kotlin.nj2k.callOn
+import org.jetbrains.kotlin.nj2k.tree.*
+
+class StreamConversion(context: NewJ2kConverterContext) : RecursiveConversion(context) {
+
+    context(KtAnalysisSession)
+    override fun applyToElement(element: JKTreeElement): JKTreeElement {
+        if (element !is JKQualifiedExpression) return recurse(element)
+        if (element.selector !is JKCallExpressionImpl) return recurse(element)
+
+        val target = (element.selector as JKCallExpressionImpl).identifier.fqName
+
+        // TODO: Handle other cases too.
+        return recurse(
+            when (target) {
+                "java.lang.String.lines" -> convertStringLines(element)
+                else -> element
+            }.withFormattingFrom(element)
+        )
+    }
+
+    private fun convertStringLines(element: JKQualifiedExpression): JKTreeElement {
+        val lineSequence = "kotlin.text.lineSequence"
+        val asStream = "kotlin.streams.asStream"
+
+        context.importStorage.addImport(asStream)
+
+        return element.receiver.copyTreeAndDetach()
```

Comment:
It's better to avoid copying elements, if possible. See `plugins/kotlin/docs/j2k/j2k.md`